src/firebolt/async_db/cursor.py

```diff
@@ -383,8 +400,12 @@ async def _do_execute(
 
                     # set parameter passed validation
                     self._set_parameters[query.name] = query.value
+                elif async_execution:
+                    resp = await self._async_api_request(query)
+                    await self._raise_if_error(resp)
+                    self._query_id = self._query_id_from_response_async(resp)
                 else:
-                    resp = await self._api_request(query, set_parameters)
+                    resp = await self._api_request(query, {})
```

Comment:
nit: make `set_parameters` argument in `_api_request` optional, having None by default

Reply:
Done.